optimum/onnxruntime/base.py

```diff
@@ -115,39 +115,40 @@ def __init__(
         if len(self.key_value_output_names) == 0:
             self.key_value_output_names = [key for key in self.output_names if "key_values" in key]
 
-        if len(self.key_value_output_names) == 0:
-            raise RuntimeError("Could not find the past key values in the provided model.")
-
-        # Attributes useful when computing the past key/values output shapes.
-        self.expected_key_symbolic_shape = None
-        self.expected_value_symbolic_shape = None
-        for output in self.session.get_outputs():
-            if ".key" in output.name:
-                self.expected_key_symbolic_shape = output.shape
-            elif ".value" in output.name:
-                self.expected_value_symbolic_shape = output.shape
-            # To handle the old case when past_key_values were following the format: past_key_values_{idx}
-            elif "key_values" in output.name:
-                if self.expected_key_symbolic_shape is None:
+        if self.parent_model.use_cache is True or len(self.key_value_output_names) != 0:
```

Comment:
```suggestion
        if len(self.key_value_output_names) != 0:
```
Since the next if raises an error if not?

Reply:
This I added for `TrOCR` because when use_cache = False, currently it does not output the `past_key_values`. So this will skip this and user can do inference with use_cache = False